Packs/TOPdesk/Integrations/TOPdesk/TOPdesk.py

```diff
@@ -730,6 +811,57 @@ def get_incidents_list(
     return incidents
 
 
+def get_assets_list(
+    client: Client,
+    args: dict[str, Any] = {},
+) -> list[dict[str, Any]]:
+    """Get list of Assets from TOPdesk.
+
+    Args:
+        client: The client from which to make the requests.
+        args: might contain new style query or other old style arguments.
+
+    Return list of incidents got from the API.
+    """
+
+    page_size = args.get("page_size", None)
+    page_size = arg_to_number(args.get("page_size", 50))
+    start = args.get("start", 0)
+    query = args.get("filter", None)
+    search_term = args.get("search_term", None)
+    archived = args.get("archived", None)
+    pagination = False
+    # If the page size is 0, we will fetch all data
+    if page_size == 0:
+        pagination = True
+        page_size = 1000
+    else:
+        pagination = False
+    assets_list = []
+    while True:
+        assets = client.get_asset_list_with_query(
+            list_type="assets",
+            start=start,
+            page_size=page_size,
+            query=query,
+            search_term=search_term,
+            archived=archived,
+            fields=args.get("fields", None),
+        )
+        if len(assets["dataSet"]) < 1:
+            break
+        else:
+            assets_list += assets["dataSet"]
+            start += page_size
+        assets_list.extend(assets["dataSet"])
+        start += page_size
```

Comment:
Same here, code got duplicated
```suggestion

        assets_list.extend(assets["dataSet"])
        start += page_size
```